### PR TITLE
[reasoner] Look at equivalent concepts in both source and destination...

### PR DESCRIPTION
**Pull request by AAAlinaaa**
... equivalency checks while switching to equivalent concepts.

### CODE REVIEW

File: snomed/com.b2international.snowowl.snomed.reasoner.server/src/com/b2international/snowowl/snomed/reasoner/server/classification/EquivalentConceptMerger.java [hunk 1]
```diff
@@ -375,4 +369,26 @@ private void removeDeprecatedRelationships(final Concept conceptToKeep, final Co
 		}
 	}
 	
+	private boolean isEquivalentInboundRelationship(final Multimap<Concept, Concept> equivalentConcepts, Relationship relationshipToRemove, Relationship relationshipToKeep) {
+		Concept sourceToKeep = relationshipToKeep.getSource();
+		Concept sourceToRemove = relationshipToRemove.getSource();
+		
+		equivalentConcepts.get(sourceToKeep);
```
Comment (apeteri) [inline snomed/com.b2international.snowowl.snomed.reasoner.server/src/com/b2international/snowowl/snomed/reasoner/server/classification/EquivalentConceptMerger.java]: This seems to be a no-op.

File: snomed/com.b2international.snowowl.snomed.reasoner.server/src/com/b2international/snowowl/snomed/reasoner/server/classification/EquivalentConceptMerger.java [hunk 2]
```diff
@@ -375,4 +369,26 @@ private void removeDeprecatedRelationships(final Concept conceptToKeep, final Co
...
+				&& (sourceToRemove.equals(sourceToKeep) || equivalentConcepts.get(sourceToKeep).contains(sourceToRemove))
+				&& relationshipToRemove.getCharacteristicType().equals(relationshipToKeep.getCharacteristicType())
+				&& relationshipToRemove.getModifier().equals(relationshipToKeep.getModifier());
+	}
+	
+	private boolean isEquivalentOutboundRelationship(final Multimap<Concept, Concept> equivalentConcepts, Relationship relationshipToRemove, Relationship relationshipToKeep) {
+		Concept destinationToKeep = relationshipToKeep.getDestination();
+		Concept destinationToRemove = relationshipToRemove.getDestination();
+		
+		equivalentConcepts.get(destinationToKeep);
```
Comment (apeteri) [inline snomed/com.b2international.snowowl.snomed.reasoner.server/src/com/b2international/snowowl/snomed/reasoner/server/classification/EquivalentConceptMerger.java]: Same here; please remove the two statements if they don't have any effect.

File: snomed/com.b2international.snowowl.snomed.reasoner.server/src/com/b2international/snowowl/snomed/reasoner/server/classification/EquivalentConceptMerger.java [hunk 3]
```diff
@@ -110,7 +110,7 @@ public void fixEquivalencies() {
 		final SnomedDeletionPlan deletionPlan = new SnomedDeletionPlan();
 		for (final Concept conceptToKeep : equivalentConcepts.keySet()) {
 			final Collection<Concept> conceptsToRemove = equivalentConcepts.get(conceptToKeep);
-			switchToEquivalentConcept(conceptToKeep, conceptsToRemove, inboundRelationshipMap);
+			switchToEquivalentConcept(conceptToKeep, conceptsToRemove, inboundRelationshipMap, equivalentConcepts);
```
Comment (apeteri) [inline snomed/com.b2international.snowowl.snomed.reasoner.server/src/com/b2international/snowowl/snomed/reasoner/server/classification/EquivalentConceptMerger.java:113]: Do we need the extra `Multimap` argument for this check? You already have concept to keep, which will be the same source/destination for all outbound/inbound relationships, and we only need to compare the destination/source against the value for this concept in the `Multimap`, ie. `conceptsToRemove`.

**Review by apeteri — CHANGES_REQUESTED** [review]

File: snomed/com.b2international.snowowl.snomed.reasoner.server/src/com/b2international/snowowl/snomed/reasoner/server/classification/EquivalentConceptMerger.java (hunk 3 above)
Comment (apeteri) [reply snomed/com.b2international.snowowl.snomed.reasoner.server/src/com/b2international/snowowl/snomed/reasoner/server/classification/EquivalentConceptMerger.java:113]: Following an off-line discussion, the answer is yes.

**Review by apeteri — APPROVED** [review]

File: snomed/com.b2international.snowowl.snomed.reasoner.server/src/com/b2international/snowowl/snomed/reasoner/server/classification/EquivalentConceptMerger.java (hunk 3 above)
Comment (cmark) [reply snomed/com.b2international.snowowl.snomed.reasoner.server/src/com/b2international/snowowl/snomed/reasoner/server/classification/EquivalentConceptMerger.java:113]: A comment would be nice to have for future maintainers. 😄

**Review by cmark — APPROVED** [review]
LGTM!

**Pull request merged**